android/java/org/chromium/chrome/browser/crypto_wallet/util/BalanceHelper.java

```diff
@@ -228,36 +228,36 @@ public static void getP3ABalances(WeakReference<BraveWalletBaseActivity> activit
                                 AsyncUtils.GetBlockchainTokensBalancesResponseContext>();
 
                 for (int coinType : Utils.P3ACoinTypes) {
-                    keyringService.getKeyringInfo(AssetUtils.getKeyringForCoinType(coinType), keyringInfo -> {
-                        for (NetworkInfo network : sortedNetworks.get(coinType)) {
-                            TokenUtils.getUserOrAllTokensFiltered(braveWalletService,
-                                    blockchainRegistry, network, coinType, TokenUtils.TokenType.ALL,
-                                    true, tokens -> {
-                                        AsyncUtils.GetNativeAssetsBalancesResponseContext
-                                                getNativeAssetsBalancesContext =
-                                                new AsyncUtils
-                                                        .GetNativeAssetsBalancesResponseContext(
+                    for (NetworkInfo network : sortedNetworks.get(coinType)) {
```

Comment:
Could we spit into several methods in this PR or log an issue for refactoring since call stack is too deep and it is hard to read it?

Reply:
Yes! If not too time-consuming I'm going to try to split this part into several methods and make it more readable.

Reply:
`BalanceHelper` refactoring addressed in commit 👉 [`3829867` (#18144)](https://github.com/brave/brave-core/pull/18144/commits/3829867d545e5c1bba70c10741dd679a5763d9d8)